var/spack/repos/builtin/packages/py-pytest/package.py

```diff
@@ -49,6 +50,7 @@ class PyPytest(PythonPackage):
     depends_on('py-attrs@17.2.0:', when='@3.3:3.4', type=('build', 'run'))
     depends_on('py-iniconfig', when='@6.0:', type=('build', 'run'))
     depends_on('py-packaging', when='@4.6:', type=('build', 'run'))
+    depends_on('py-pluggy@0.12:1', when='@6.2:', type=('build', 'run'))
     depends_on('py-pluggy@0.12:0', when='@4.6:', type=('build', 'run'))
```

Comment:
```suggestion
    depends_on('py-pluggy@0.12:0', when='@4.6:6.1', type=('build', 'run'))
```
Otherwise the same constraint will be applied to 6.2+